core/unix/os.c

```diff
@@ -838,6 +843,72 @@ get_uname(void)
 #endif
 }
 
+#if defined(LINUX) && !defined(DR_HOST_NOT_TARGET)
+static int
+make_failing_clone3_syscall()
+{
+#    if defined(ARM)
+    /* TODO i#5221: Implement for ARM. */
+#    else
+    long result;
+    /* We know that clone3 fails with EINVAL with these args. */
+    ulong clone_args_size = 0;
+    void *clone_args = NULL;
+#        ifdef X86
+#            ifdef X64
+    asm volatile("mov %[sys_clone3], %%rax\n\t"
```

Comment:
Replace all of this assembly with a single dynamorio_syscall() (i.e., we already have per-arch asm to make syscalls; no need to duplicate it here).

Reply:
Done.